### ISSUE TITLE
Unable to add custom RPC with the same chainID

### ISSUE THREAD

**Issue by 0xGitGuy** [RabbyHub/Rabby#2276]
This issue was somewhat discuss before https://github.com/RabbyHub/Rabby/issues/261

I get an error when attempting to add a second RPC for ethereum. 

The error: Chain already integrated by Rabby Wallet

Why can we only have one RPC per chain? Metamask supports multiple RPC for the same chain. This is not only useful for dev work but also as a quick backup for switching between RPC provider.

**Comment by 0xGitGuy** [comment]
"Modify RPC URL" feature is not a solution at all. Still could only have 1 RPC provider for a given chain.

Is there any particular reason for Rabby to force users to only have 1 RPC per chain since adding custom RPC intentionally checks for duplication chainID?

**Comment by xtools-at** [comment]
it kinda makes sense that the chain id is the "unique identifier" in this context and may be used for chain selection.

what's your use case? connecting to the same chain with different rpcs?

**Comment by 0xGitGuy** [comment]
yes, same chain diff rpc. mevblock has different variants of rpc that produces different results (revert vs no revert). The public rpc produces the pending mempool txn. I use them for various trading and dev mainnet testing purposes. Sticking to 1 rpc means i have to toggle between rabby and metamask as a tradeoff, not a pleasant UX

**Comment by liqlootdev** [comment]
> it kinda makes sense that the chain id is the "unique identifier" in this context and may be used for chain selection.
> 
> what's your use case? connecting to the same chain with different rpcs?

What's the use case? of having multiple rpc options on a custom network???
Its obviously needed for when one goes down, which is extremely common in the context of using custom networks.

**Comment by shrpne** [comment]
> what's your use case? connecting to the same chain with different rpcs?

You may want to protect one transaction with Flashbots, use low gas for the second transaction with GasHawk, and send third transaction directly to regular RPC

**Comment by 11qu1d** [comment]
Was looking to switch from MetaMask, kinda disappointing this basic feature is not available 4 years after it was initially requested.
